libp2p/src/lib.rs

```diff
@@ -65,7 +65,7 @@ pub use libp2p_dns as dns;
 #[doc(inline)]
 pub use libp2p_floodsub as floodsub;
 #[cfg(feature = "gossipsub")]
-#[cfg(not(target_os = "unknown"))]
+#[cfg(any(not(target_os = "unknown"), target_arch = "wasm32"))]
```

Comment:
I am not sure what was the target for `unknown`, can we even remove this condition completely?

Reply:
I would not know why its there and might be safe to remove just by scanning through the code. It could have been used to prevent using `wasm32-unknown-unknown` and similar targets with a unknown os.

Reply:
If it compiles properly, we should be able to just remove this `cfg` everywhere :)